bakerydemo/templates/base/basic_auth.html

```diff
@@ -0,0 +1,16 @@
+{% load wagtailadmin_tags %}
+
+<title>Authentication Required</title>
+<style>
+    h1 { font-size: 50px; }
+    body { font: 20px sans-serif; color: #333; }
+    article { display: block; text-align: center; margin: 0 auto; }
+    a { color: #dc8100; text-decoration: none; }
+    a:hover { color: #333; text-decoration: none; }
+</style>
+
+<article>
+    <h1>Authentication Required</h1>
+    <p>You must login to view this site</p>
+    <h3>Looking for Wagtail's demo instance? <a href="https://demo.wagtail.org">Click here</a>.</h3>
```

Comment:
Out of curiosity, what's the motivation for including basic auth support as part of this PR, and in what situation do you envisage people stumbling upon a password-protected bakerydemo instance while looking for the demo site? I feel like I've missed a memo about how we intend to use this :-)

Reply:
It's a great question I could have explained more. I included it here so that we can easily add protection to the current "latest" demo environment, moving it from a public instance to more of an internal (wagtail core etc) development tool. Putting basic auth adds a level of protection to it which we want, but in case people still have the link saved somewhere, this means they'll receive a link to the new demo instance. Note they only see the page if they fail the prompt a couple times (browser limitation, nothing we can do).